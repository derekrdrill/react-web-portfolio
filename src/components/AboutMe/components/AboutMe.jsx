import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, List, ListItem } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ABOUT_ME_TEXT } from '../constants/ABOUT_ME_TEXT';
import bitmojiChillinWithBirds from '../../../assets/bitmoji_chillinWithBirds1.png';

export const AboutMe = ({ id }) => {
  // test change

  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <AboutMeContainer darkMode={darkMode} id={id} fluid>
        <Row>
          <Col className='py-3'>
            <AboutMeTitle className='text-center' component='h1' darkMode={darkMode} variant='h3'>
              ABOUT ME
            </AboutMeTitle>
          </Col>
        </Row>
        <Row>
          <AboutMeTextContainer
            darkMode={darkMode}
            xs={{ span: 10, order: 'last', offset: 1 }}
            lg={{ span: 5, order: 'first', offset: 1 }}
          >
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[1].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[2].text}
            </AboutMeText>
            <List style={{ color: 'white' }}>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[3].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[4].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[5].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[6].text}</AboutMeListItem>
            </List>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[7].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[8].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[9].text}
            </AboutMeText>
          </AboutMeTextContainer>
          <AboutMeBitmojiContainer xs={12} lg={6}>
            <img src={bitmojiChillinWithBirds} alt='Chillin with birds' />
          </AboutMeBitmojiContainer>
        </Row>
      </AboutMeContainer>
    </>
  );
};

const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  },
}));

export const AboutMeContainer = styled(Container)(({ darkMode }) => ({
  padding: '30px 50px 200px 50px',
  borderTop: '2px solid transparent',
  borderImage: darkMode
    ? 'linear-gradient(to right, rgba(248, 184, 255, 1), skyblue, gainsboro)'
    : 'linear-gradient(to right, rgba(176, 52, 197, 1), rgba(86, 206, 210, 1))',
  borderImageSlice: 1,
}));

export const AboutMeTitle = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'gainsboro' : 'darkslategray',
}));

export const AboutMeTextContainer = styled(Col)(({ darkMode }) => ({
  padding: 40,
  height: 550,
  overflowY: 'auto',
  border: '1px dashed #383838',
  borderRadius: 2,
  backgroundColor: darkMode ? '#303030' : '#eee5c1',
}));

export const AboutMeBitmojiContainer = styled(Col)({
  textAlign: 'center',
  paddingTop: 75,
  paddingBottom: 20,
});

export const AboutMeText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
  marginBottom: 15,
}));

export const AboutMeListItem = styled(ListItem)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
}));
