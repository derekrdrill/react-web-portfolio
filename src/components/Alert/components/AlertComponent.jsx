import React, { useContext } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Alert, AlertTitle } from '@mui/material';
import { AlertContext } from '../context/AlertContext';

export const AlertComponent = () => {
  const { alert, fadeOut } = useContext(AlertContext);

  return (
    alert && (
      <StyledAlert severity={alert.type} $fadeOut={fadeOut}>
        <AlertTitle>{alert.title}</AlertTitle>
        {alert.msg}
      </StyledAlert>
    )
  );
};

const fadeOut = keyframes({
  '0%': {
    opacity: 1,
  },
  '100%': {
    opacity: 0,
  },
});

const StyledAlert = styled(Alert)`
  animation-name: ${props =>
    props.$fadeOut
      ? css`
          ${fadeOut}
        `
      : 'none'};
  animation-duration: 500ms;
  animation-direction: linear;
`;
