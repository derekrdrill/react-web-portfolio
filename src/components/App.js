import React from 'react';

const AboutMe = React.lazy(() => import('./AboutMe/components/AboutMe'));
const ConnectWithMe = React.lazy(() => import('./ConnectWithMe/components/ConnectWithMe'));
const Homepage = React.lazy(() => import('./Homepage/Homepage'));
const Projects = React.lazy(() => import('./Projects/components/Projects'));

const App = () => (
  <div className='App'>
    <Homepage id='home' />
    <AboutMe id='about-me' />
    <Projects id='projects' />
    <ConnectWithMe id='connect-with-me' />
  </div>
);

export default App;
