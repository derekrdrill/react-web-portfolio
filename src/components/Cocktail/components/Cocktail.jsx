import React from 'react';
import PropTypes from 'prop-types';
import styled, { createGlobalStyle } from 'styled-components';

import { AlertProvider } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import CocktailDetail from './CocktailDetail';
import CocktailSearch from './CocktailSearch';

const Cocktail = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <AlertProvider>
      <PageStyle darkMode={darkMode} />
      <CocktailSearch />
      <CocktailDetail />
    </AlertProvider>
  );
};

Cocktail.propTypes = {};

export default Cocktail;

const PageStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#1B0008' : '#FBAF93',
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
    color: darkMode ? 'beige' : '#464646',
  },
}));
