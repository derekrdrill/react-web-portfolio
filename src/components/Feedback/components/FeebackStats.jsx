import React, { useContext } from 'react';
import styled from 'styled-components';
import { FeedbackContext } from '../context/FeedbackContext';

export const FeedbackStats = () => {
  const { feedbackItems } = useContext(FeedbackContext);

  let average =
    feedbackItems.reduce((acc, current) => {
      return acc + current.rating;
    }, 0) / feedbackItems.length;

  return (
    <FeedbackStatsDiv>
      <h4>{feedbackItems.length} reviews</h4>
      <h4>Average rating: {isNaN(average) ? 0 : average % 1 > 0 ? average.toFixed(1) : average}</h4>
    </FeedbackStatsDiv>
  );
};

const FeedbackStatsDiv = styled.div({
  alignItems: 'center',
  color: '#fff',
  display: 'flex',
  justifyContent: 'space-between',
  margin: '20px 30px',
});
