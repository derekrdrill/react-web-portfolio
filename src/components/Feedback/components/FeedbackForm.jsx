import React, { useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Button, FormControl, FormControlLabel, Radio, RadioGroup, TextField, Typography } from '@mui/material';
import { Card } from '../../Card/Card';
import { FeedbackContext } from '../context/FeedbackContext';

export const FeedbackForm = ({ maxRank }) => {
  const menuItems = new Array(maxRank + 1).fill(null);
  const { addFeedbackItem, feedbackEdit, updateFeedbackItem } = useContext(FeedbackContext);

  const [rating, setRating] = useState(null);
  const [review, setReview] = useState('');

  const handleSetRating = (e, value) => setRating(Number(value));
  const handleSetReview = e => setReview(e.currentTarget.value);
  const handleSubmitReview = () => {
    if (!feedbackEdit.edit) {
      addFeedbackItem({
        rating: rating,
        text: review,
      });
    } else {
      updateFeedbackItem({
        _id: feedbackEdit.item.id,
        rating: rating,
        text: review,
      });
    }

    setRating(null);
    setReview('');
  };

  useEffect(() => {
    if (feedbackEdit.edit) {
      setReview(feedbackEdit.item.text);
      setRating(feedbackEdit.item.rating);
    }
  }, [feedbackEdit]);

  return (
    <Card>
      <h2>Feedback Form</h2>
      <Typography component='p' variant='subtitle1'>
        Please rate your experience
      </Typography>
      <FormControl fullWidth>
        <RadioGroup onChange={handleSetRating}>
          <RadioButtonsContainer>
            {menuItems.map((menuItem, menuItemKey) => (
              <div key={menuItemKey}>
                <RatingRadioControl
                  checked={menuItemKey === rating}
                  control={<Radio />}
                  label={menuItemKey}
                  value={menuItemKey}
                />
              </div>
            ))}
          </RadioButtonsContainer>
        </RadioGroup>
      </FormControl>
      <TextField
        fullWidth
        label={feedbackEdit.edit ? 'Edit review' : 'Write a review'}
        minRows={2}
        multiline
        onChange={handleSetReview}
        placeholder={feedbackEdit.edit ? 'Edit review' : 'Write a review'}
        value={review}
      />
      <ButtonContainer>
        <Button
          disabled={!review || typeof rating !== 'number'}
          onClick={handleSubmitReview}
          size='small'
          variant='outlined'
        >
          {feedbackEdit.edit ? 'Update ' : 'Submit '}
          Review
        </Button>
      </ButtonContainer>
    </Card>
  );
};

FeedbackForm.propTypes = {
  maxRank: PropTypes.number.isRequired,
};

FeedbackForm.defaultProps = {
  maxRank: 10,
};

const ButtonContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-end',
  padding: '15px 0',
});

const RadioButtonsContainer = styled.div({
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'center',
  margin: '40px 30px',
});

const RatingRadioControl = styled(FormControlLabel)(({ checked }) => [
  {
    '.MuiRadio-root': {
      display: 'none',
    },
    ':hover': {
      backgroundColor: '#FF77BC',
      color: 'beige',
    },
    backgroundColor: 'lavender',
    borderRadius: 50,
    color: '#333',
    minWidth: 12,
    padding: '25px 30px',
    cursor: 'pointer',
  },
  checked && {
    ':hover': {
      backgroundColor: '#be7dff',
    },
    backgroundColor: '#be7dff',
    color: 'beige',
  },
]);
