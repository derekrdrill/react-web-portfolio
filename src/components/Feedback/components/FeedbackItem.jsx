import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Button, Typography } from '@mui/material';
import { Card } from '../../Card/Card';
import { BasicModal } from '../../Modals/BasicModal';
import { FeedbackContext } from '../context/FeedbackContext';
import { FaTimes, FaEdit } from 'react-icons/fa';

export const FeedbackItem = ({ bgColorNumDisplay, id, rating, text }) => {
  const { deleteFeedbackItem, setFeedbackEditItems } = React.useContext(FeedbackContext);
  const [modalOpen, setModalOpen] = useState(false);
  const handleModalOpen = () => setModalOpen(true);
  const handleModalClose = () => setModalOpen(false);

  return (
    <Card>
      <NumberDisplay bgColor={bgColorNumDisplay}>
        <Typography variant='subtitle' component='h6'>
          {rating}
        </Typography>
      </NumberDisplay>
      <ActionButtonsDisplay>
        <ActionsButtonContainer>
          <Button
            color='secondary'
            fullWidth
            onClick={() =>
              setFeedbackEditItems(
                {
                  id: id,
                  text: text,
                  rating: rating,
                },
                true,
              )
            }
          >
            <FaEdit color='purple' />
          </Button>
        </ActionsButtonContainer>
        <ActionsButtonContainer>
          <Button color='secondary' fullWidth onClick={handleModalOpen}>
            <FaTimes color='purple' />
          </Button>
        </ActionsButtonContainer>
      </ActionButtonsDisplay>
      <TextDisplay>
        <Typography variant='subtitle1' component='p'>
          {text}
        </Typography>
      </TextDisplay>
      <BasicModal
        cancelButtonText='I changed my mind'
        handleClose={handleModalClose}
        handleSubmit={() => deleteFeedbackItem(id)}
        open={modalOpen}
        submitButtonText='Yes, delete it'
      >
        <Typography paragraph>Are you sure you want to delete this feedback record?</Typography>
      </BasicModal>
    </Card>
  );
};

FeedbackItem.propTypes = {
  bgColorNumDisplay: PropTypes.string,
  id: PropTypes.string,
  rating: PropTypes.number,
  text: PropTypes.string,
};

export const NumberDisplay = styled.div(({ bgColor }) => ({
  position: 'relative',
  background: '#f4f4f4',
  width: 50,
  height: 50,
  padding: 4,
  textAlign: 'center',
  borderRadius: '50%',
  fontSize: 30,
  border: '1px #eee solid',
  transition: '0.3s',
  color: '#FFF',
  backgroundColor: bgColor,
  transform: 'translate(-60px, -50px)',
  '.MuiTypography-root': {
    paddingTop: 12,
  },
}));

const ActionsButtonContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-end',
  maxWidth: 50,
});

const ActionButtonsDisplay = styled.div({
  position: 'absolute',
  top: 10,
  right: 1,
  cursor: 'pointer',
  background: 'none',
  border: 'none',
  display: 'flex',
  justifyContent: 'flex-end',
});

const TextDisplay = styled.div({
  transform: 'translate(0px, -40px)',
});
