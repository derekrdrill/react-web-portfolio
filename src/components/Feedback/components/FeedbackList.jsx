import React from 'react';
import { FeedbackContext } from '../context/FeedbackContext';
import { FeedbackItem } from './FeedbackItem';

export const FeedbackList = () => {
  const { feedbackItems } = React.useContext(FeedbackContext);

  return (
    <>
      {feedbackItems &&
        feedbackItems.map(feedbackItem => (
          <FeedbackItem
            bgColorNumDisplay='#FF77BC'
            id={feedbackItem._id}
            key={feedbackItem._id}
            rating={feedbackItem.rating}
            text={feedbackItem.text}
          />
        ))}
    </>
  );
};
