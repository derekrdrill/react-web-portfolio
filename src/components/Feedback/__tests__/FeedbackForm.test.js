import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { FeedbackForm, RatingRadioControl } from '../components/FeedbackForm';
import {
  getFeedbackReviewLabel,
  getSubmitButtonLabel,
  isReviewButtonDisabled,
} from '../components/FeedbackForm';

configure({ adapter: new Adapter() });

const renderer = new ShallowRenderer();

describe('Feedback Form tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({
      addFeedbackItem: '',
      feedbackEdit: '',
      updateFeedbackItem: '',
    });

    renderer.render(<FeedbackForm />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders RatingRadioControl styled component correctly', () => {
    const ratingRadioControl = shallow(<RatingRadioControl />);
    const ratingRadioControlChecked = shallow(<RatingRadioControl checked />);

    expect(ratingRadioControl.props().className).toEqual('sc-fqkvVR jXQJxZ');
    expect(ratingRadioControlChecked.props().className).toEqual('sc-fqkvVR doYZOv');
  });

  it('runs getFeedbackReviewLabel correctly', () => {
    expect(getFeedbackReviewLabel(true)).toEqual('Edit review');
    expect(getFeedbackReviewLabel(false)).toEqual('Write a review');
  });

  it('runs getSubmitButtonLabel correctly', () => {
    expect(getSubmitButtonLabel(true)).toEqual('Update ');
    expect(getSubmitButtonLabel(false)).toEqual('Submit ');
  });

  it('runs isReviewButtonDisabled correctly', () => {
    expect(isReviewButtonDisabled(true, 1)).toEqual(false);
    expect(isReviewButtonDisabled(false, 1)).toEqual(true);
    expect(isReviewButtonDisabled(false, '1')).toEqual(true);
  });
});
