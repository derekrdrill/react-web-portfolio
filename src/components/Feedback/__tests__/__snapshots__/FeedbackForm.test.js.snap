// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Form tests renders correctly 1`] = `
<Card
  darkMode={false}
  padding={true}
>
  <h2>
    Feedback Form
  </h2>
  <ForwardRef(Typography)
    component="p"
    variant="subtitle1"
  >
    Please rate your experience
  </ForwardRef(Typography)>
  <ForwardRef(FormControl)
    fullWidth={true}
  >
    <ForwardRef(RadioGroup)
      onChange={[Function]}
    >
      <styled.div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={0}
            value={0}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={1}
            value={1}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={2}
            value={2}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={3}
            value={3}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={4}
            value={4}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={5}
            value={5}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={6}
            value={6}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={7}
            value={7}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={8}
            value={8}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={9}
            value={9}
          />
        </div>
        <div>
          <Styled(Component)
            checked={false}
            control={<ForwardRef(Radio) />}
            label={10}
            value={10}
          />
        </div>
      </styled.div>
    </ForwardRef(RadioGroup)>
  </ForwardRef(FormControl)>
  <ForwardRef(TextField)
    fullWidth={true}
    label="Write a review"
    minRows={2}
    multiline={true}
    onChange={[Function]}
    placeholder="Write a review"
    value=""
  />
  <styled.div>
    <ForwardRef(Button)
      disabled={true}
      onClick={[Function]}
      size="small"
      variant="outlined"
    >
      Submit 
      Review
    </ForwardRef(Button)>
  </styled.div>
</Card>
`;
